src/main/scala/rocket/icache.scala

```diff
@@ -128,7 +128,7 @@ class ICache(latency: Int)(implicit p: Parameters) extends CoreModule()(p) with
       io.resp.bits.datablock := Mux1H(s1_tag_hit, s1_dout)
       io.resp.valid := s1_hit
     case 2 =>
-      val s2_hit = RegEnable(s1_hit, !stall)
+      val s2_hit = RegEnable(s1_hit, Bool(false), !stall)
```

Comment:
Is there a general rule/ spec for rocket-chip  like "Clock must run for at
least 2 cycles in reset"? These would be pretty important to write down...

On Thu, Aug 25, 2016 at 6:07 PM, Andrew Waterman <notifications@github.com>
wrote:

> In src/main/scala/rocket/icache.scala
> <https://github.com/ucb-bar/rocket-chip/pull/230#discussion_r76351277>:
>
> > @@ -128,7 +128,7 @@ class ICache(latency: Int)(implicit p: Parameters) extends CoreModule()(p) with
> >        io.resp.bits.datablock := Mux1H(s1_tag_hit, s1_dout)
> >        io.resp.valid := s1_hit
> >      case 2 =>
> > -      val s2_hit = RegEnable(s1_hit, !stall)
> > +      val s2_hit = RegEnable(s1_hit, Bool(false), !stall)
>
> Likewise, this should become false after 2 cycles in reset, since stall=0
> and s1_hit=0 during reset.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ucb-bar/rocket-chip/pull/230/files/c17bf28a9fa139f8ec37a867d03884fa06e1ab3e#r76351277>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARCAJNanLMT5EplcKompsi5TOlny__sIks5qjjxFgaJpZM4JtoX6>
> .
>



-- 
Megan A. Wachs
Engineer | SiFive, Inc
300 Brannan St, Suite 403
San Francisco, CA  94107
megan@sifive.com